crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -12,3 +26,751 @@ pub enum RoomNotificationMode {
+    }
+
+    #[async_test]
+    async fn test_delete_rules_api_error() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+        let room_id = get_test_room_id();
+
+        Mock::given(method("DELETE")).respond_with(ResponseTemplate::new(500)).mount(&server).await;
+
+        let notification_settings = client.notification_settings().await;
+        let mut rules = notification_settings.rules().await;
+        _ = rules.insert_room_rule(RuleKind::Room, &room_id, true).unwrap();
+        notification_settings.set_ruleset(&rules.ruleset).await;
+
+        let rules_to_delete = &[(RuleKind::Room, room_id.to_string())];
+        assert_eq!(
+            Err(NotificationSettingsError::UnableToRemovePushRule),
+            notification_settings.delete_rules(rules_to_delete, &[]).await
+        );
+    }
+
+    #[async_test]
+    async fn test_delete_rules_not_present() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+        let room_id = get_test_room_id();
+
+        Mock::given(method("DELETE")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+
+        let notification_settings = client.notification_settings().await;
+
+        let rules_to_delete = &[(RuleKind::Room, room_id.to_string())];
+        assert_eq!(
+            Err(NotificationSettingsError::UnableToRemovePushRule),
+            notification_settings.delete_rules(rules_to_delete, &[]).await
+        );
+    }
+
+    #[async_test]
+    async fn test_delete_user_defined_room_rules() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+        let room_id_a = RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap();
+        let room_id_b = RoomId::parse("!BBBbBBBBBbbBBbbbbb:matrix.org").unwrap();
+
+        Mock::given(method("DELETE")).respond_with(ResponseTemplate::new(200)).mount(&server).await;
+
+        let notification_settings = client.notification_settings().await;
+
+        // Insert some initial rules
+        let mut rules = notification_settings.rules().await;
+        _ = rules.insert_room_rule(RuleKind::Room, &room_id_a, true).unwrap();
+        _ = rules.insert_room_rule(RuleKind::Override, &room_id_a, true).unwrap();
+        _ = rules.insert_room_rule(RuleKind::Room, &room_id_b, true).unwrap();
+        _ = rules.insert_room_rule(RuleKind::Override, &room_id_b, true).unwrap();
+        notification_settings.set_ruleset(&rules.ruleset).await;
+
+        // Delete all user defined rules for room_id_a
+        notification_settings.delete_user_defined_room_rules(&room_id_a).await.unwrap();
+
+        // Only the rules for room_id_b should remain
+        let remaining_rules = notification_settings.get_custom_rules_for_room(&room_id_b).await;
+        assert_eq!(remaining_rules.len(), 2);
```

Comment:
Can you also check that rules for `room_id_a` are *not* there?

Reply:
done.